src/util-base64.c

```diff
@@ -196,7 +196,7 @@ Base64Ecode DecodeBase64(uint8_t *dest, uint32_t dest_size, const uint8_t *src,
     {                                                                                              \
         uint32_t consumed_bytes = 0, num_decoded = 0;                                              \
         uint8_t dst[dest_size];                                                                    \
-        Base64Ecode code = DecodeBase64(dst, strlen(fin_str), (const uint8_t *)src, strlen(src),   \
+        Base64Ecode code = DecodeBase64(dst, dest_size, (const uint8_t *)src, strlen(src),         \
```

Comment:
This looks wrong to me, it should have remained `strlen(fin_str)` and `strlen(fin_str)` should be used in ` uint8_t dst[dest_size];  `

Reply:
Could you please tell why is it wrong? I changed it because the callers are responsible for sending any dest_size, so, the tests here would also be able to replicate the same..

Reply:
Maybe wrong is not the right term.

But we want `dest_size` to be equal to `strlen(fin_str)` and so we should not use a second argument/variable